docs/authentication.md

```diff
@@ -0,0 +1,211 @@
+# Authentication Functions
+
+<!-- Glee relies on functions to execute your business logic. Functions are files that export a default async Node.js function: -->
+
+Authentication in Glee can be done using authentication functions. Authentication functions are files that export either one or both of the `clientAuth` and `serverAuth` Node.js functions:
+
+```js
+/* websocket.js */
+
+export async function serverAuth({ authProps, done }) {
+  //server auth logic
+}
+
+export async function clientAuth({ parsedAsyncAPI, serverName }) {
+  //client auth logic
+}
+```
+
+Glee looks for authentication files in the `auth` directory, the name of the authentication file should be the name of the targeted server that the authentication logic should work for.
```

Comment:
Just like `functions` and `lifecycle` folder, I should be able to change the location of my `auth` function as well.
please add this option in:
https://github.com/asyncapi/glee/blob/d83701dec664695c9928999fee30f4453d655d83/docs/config-file.md?plain=1#L98

also reflect the changes in https://github.com/asyncapi/glee/pull/474